mutable_tree.go

```diff
@@ -556,10 +735,78 @@ func (tree *MutableTree) SaveVersion() ([]byte, int64, error) {
 	tree.ImmutableTree = tree.ImmutableTree.clone()
 	tree.lastSaved = tree.ImmutableTree.clone()
 	tree.orphans = map[string]int64{}
+	tree.unsavedFastNodeAdditions = make(map[string]*FastNode)
+	tree.unsavedFastNodeRemovals= make(map[string]interface{})
 
 	return tree.Hash(), version, nil
 }
 
+func (tree *MutableTree) saveFastNodeVersion() error {
```

Comment:
I'm planning to look into the following:
- if removing before writing to db is faster
- if combining sorted removals with sorted additions is faster than doing them one after the other

If anyone knows, what patterns are the most efficient or how I can optimize this better, please let me know